From 8ee3219fff07c03401ec25208ef3178bcc96acd1 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:35:59 -0400
Subject: [block] blk-mq: initialize q->nr_requests after calling blk_queue_make_request()

Message-id: <1402670238-13916-72-git-send-email-snitzer@redhat.com>
Patchwork-id: 83815
O-Subject: [RHEL7.1 PATCH 071/150] blk-mq: initialize q->nr_requests after calling blk_queue_make_request()
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit eba7176826ddab1d04c51bb2d5f2bbf22865444c
Author: Jens Axboe <axboe@fb.com>
Date:   Tue May 20 15:17:27 2014 -0600

    blk-mq: initialize q->nr_requests after calling blk_queue_make_request()

    blk_queue_make_requests() overwrites our set value for q->nr_requests,
    turning it into the default of 128. Set this appropriately after
    initializing queue values in blk_queue_make_request().

    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq.c b/block/blk-mq.c
index 7b71ab1..fec8fcc 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -1623,6 +1623,11 @@ struct request_queue *blk_mq_init_queue(struct blk_mq_tag_set *set)
  if (set->timeout)
   blk_queue_rq_timeout(q, set->timeout);
 
+ /*
+  * Do this after blk_queue_make_request() overrides it...
+  */
+ q->nr_requests = set->queue_depth;
+
  if (set->ops->complete)
   blk_queue_softirq_done(q, set->ops->complete);
 
-- 
1.7.1